llama-rs/src/models/bloom.rs

```diff
@@ -0,0 +1,979 @@
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::BufReader,
+    io::{BufRead, Read, Seek, SeekFrom},
+    path::{Path, PathBuf},
+};
+
+use crate::common::{helpers::*, inference::*, load::*, model::*, token::*, vocabulary::*};
+use crate::mulf;
+
+// NOTE: Field order matters! Data is laid out in the file exactly
+// in this order.
+#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct Hyperparameters {
+    pub n_vocab: usize,
+    pub n_ctx: usize,
+    pub n_embd: usize,
+    pub n_mult: usize,
+    pub n_head: usize,
+    pub n_layer: usize,
+    pub f16_: u32,
+}
+
+// default
+pub struct Layer {
+    pub attention_norm: ggml::Tensor,
+    pub attention_norm_b: ggml::Tensor,
+    pub wo: ggml::Tensor,
+    pub wo_b: ggml::Tensor,
+    pub query_key_value: ggml::Tensor,
+    pub query_key_value_b: ggml::Tensor,
+    // normalization
+    pub ffn_norm: ggml::Tensor,
+    pub ffn_norm_b: ggml::Tensor,
+    // ff
+    pub w1: ggml::Tensor,
+    pub w1_b: ggml::Tensor,
+    pub w2: ggml::Tensor,
+    pub w2_b: ggml::Tensor,
+}
+
+/// The weights for the BLOOM model. All the mutable state is split into a
+/// separate struct `InferenceSession`.
+pub struct BLOOM {
```

Comment:
I'd go with `Bloom`:

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

https://rust-lang.github.io/api-guidelines/naming.html